marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java

```diff
@@ -257,16 +261,22 @@ public void run() {
                     stepResp = Job.withFlow(flow);
                     stepResp.withJobId(runningJobId);
                     stepResp.setCounts(successCount.get() + errorCount.get(), successCount.get(), errorCount.get(), (long) Math.ceil((double) successCount.get() / stepBatchSize), (long) Math.ceil((double) errorCount.get() / stepBatchSize));
-                    StringWriter errors = new StringWriter();
-                    e.printStackTrace(new PrintWriter(errors));
-                    stepResp.withStepOutput(errors.toString());
+                    stepResp.withStepOutput(e.getLocalizedMessage());
                     stepResp.withSuccess(false);
                     if(successCount.get() > 0) {
                         stepResp.withStatus(JobStatus.COMPLETED_WITH_ERRORS_PREFIX + stepNum);
                     }
                     else{
                         stepResp.withStatus(JobStatus.FAILED_PREFIX + stepNum);
                     }
+                    Job finalStepResp = stepResp;
+                    try {
+                        flowStatusListeners.forEach((FlowStatusListener listener) -> {
+                            listener.onStatusChanged(jobId, runningStep, JobStatus.FAILED, currPercentComplete[0], currSuccessfulEvents[0], currFailedEvents[0],
+                                runningStep.getName() + " " + Arrays.toString(finalStepResp.stepOutput.toArray()));
```

Comment:
if the step fails, percent complete is 100, it can be hardcoded. status should be  JobStatus.FAILED_PREFIX+stepNum. The status of Job as written to db at the time of this failure is JobStatus.FAILED_PREFIX+stepNum. So, at this time the job status should also be JobStatus.FAILED_PREFIX+stepNum